### DISCUSSION TITLE
Has anyone encountered "Error: write EPIPE" before?

### DISCUSSION THREAD

**Discussion by joshxyzhimself** [nodejs/undici#1161] [Q&A]
Has anyone encountered "Error: write EPIPE" before?

I do when I'm trying to POST a JSON-encoded body of 20,000 items.

Could this be a problem on the receiving server's end and not undici's?

Are there other workarounds this other than reducing the amount of items?

```
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

Error: write EPIPE
    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:98:16) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
```

**Comment by mcollina** [comment]
Does it happen all the time? Could you create a client/server couple to reproduce?

**Comment by joshxyzhimself** [comment]
I might create a repo when I can, but it went like this:

- I was running a local instance of MeiliSearch
  - https://github.com/meilisearch/MeiliSearch
- I tried importing array of 100 items, it works
  - Each object has around 8 keys, less than 20 characters each key
  - Simple POST request, request body is JSON.stringify(array_of_items)
  - My poorly-documented undici wrapper was https://github.com/repackd/modules/blob/master/packages/undici2/index.js
- I tried importing array of 20k items
  - I encountered the error below

```
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

Error: write EPIPE
    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:98:16) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
```

- Next I tried their provided client, tried the same array of 20k items, it worked
  - https://github.com/meilisearch/meilisearch-js
  - Uses https://github.com/lquixada/cross-fetch

**Comment by joshxyzhimself** [comment]
As far as I remember I also encountered the same EPIPE error when trying to POST large amount of data to my local clickhouse server instance. It's a little weird, I've seen it error at 15mb, 30mb bytelength of request body.
